File: src/medium/MaxSubArray.swift
func maxSubArray(_ nums: [Int]) -> Int {
  guard !nums.isEmpty else { return 0 }
  var maximum = nums[0]
  var ending = nums[0]
  for i in 1..<nums.count {
    ending = max(ending + nums[i], nums[i])
    maximum = max(maximum, ending)
  }
  return maximum
}

File: src/medium/max_sub_array.rs
pub fn max_sub_array(nums: Vec<i32>) -> i32 {
let mut result = nums[0];
let mut end = nums[0];
for i in 1..nums.len() {
	end = nums[i].max(end + nums[i]);
	result = result.max(end);
}
result

}